pkg/common/util/parse_resource.go

```diff
@@ -13,6 +13,8 @@ func ParseResourceEdge(resource string, operation string) (string, string, strin
 	resourceSplits := strings.Split(resource, "/")
 	if len(resourceSplits) == 3 {
 		return resourceSplits[0], resourceSplits[1], resourceSplits[2], nil
+	} else if len(resourceSplits) == 5 {
```

Comment:
I feel ill-equipment to review without any background.

Reply:
@m1093782566 , edgeController when it sends a beehive resource, it is prefixed with node/{node-id} from cloudHub to understand which node to forward that message to. CloudHub trims node/{node-id} from the resource and then forwards the message to edgeHub. In EdgeSite implementation, there is no cloudHub/edgeHub. EdgeController will directly forward the message to metaManager. So the node/{node-id} will be present. So we are adding this logic to parse that resource.

Reply:
Oh, so tricky and a disaster to extend the message topics.

Will @CindyXing's proposal: https://github.com/kubeedge/kubeedge/issues/399 help the extension issue out?

Reply:
resource is not a message topic when it used by metaManager and edgeController. It is used to pass nameSpace and nodeName of edge_core.

Yess @CindyXing 's proposal may help, but it will take time to switch all messaging to protobuf.

Reply:
Folks, this relates to the protocol between edge and cloud. In the long run, protobuf should be the direction. In current code, there are quite some hard coded numbers/index on how to parse.